sdk/core/azure-core/src/main/java/com/azure/core/util/ConfigurationBuilder.java

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.implementation.util.EnvironmentConfiguration;
+import java.util.Objects;
+
+/**
+ * Builds {@link Configuration} with external source.
+ */
+public class ConfigurationBuilder {
+    private final ConfigurationSource source;
+    private final EnvironmentConfiguration environmentConfiguration;
+    private String rootPath;
+    private Configuration sharedConfiguration;
+
+    /**
+     * Creates {@code ConfigurationBuilder}.
+     *
+     * <!-- src_embed com.azure.core.util.ConfigurationBuilder -->
+     * <pre>
+     * &#47;&#47; Creates ConfigurationBuilder with configured root path to shared properties
+     * ConfigurationBuilder configurationBuilder = new ConfigurationBuilder&#40;new SampleSource&#40;properties&#41;&#41;
+     *     .root&#40;&quot;azure.sdk&quot;&#41;; &#47;&#47; shared properties' absolute path
+     * </pre>
+     * <!-- end com.com.azure.core.util.ConfigurationBuilder -->
+     * @param source Custom {@link ConfigurationSource} containing known Azure SDK configuration properties
+     */
+    public ConfigurationBuilder(ConfigurationSource source) {
```

Comment:
Should we also want to have a no-args constructor that allows me to create a configuration by adding config values in the builder itself? We could then do something like this:
`new ConfigurationBuilder().put(key1, value1).put(key2, value2).build()`

Reply:
I like this idea! This is not something we need right away, but it seems generally useful and we can also get rid of `TestConfigurationBuilders` and simplify life for ourselves